src/org/sipdroid/sipua/PublishAgent.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2005 Luca Veltri - University of Parma - Italy
+ *
+ * This source code is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This source code is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this source code; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Author(s):
+ * Luca Veltri (luca.veltri@unipr.it)
+ */
+
+package org.sipdroid.sipua;
+
+
+import org.zoolu.sip.address.NameAddress;
+import org.zoolu.sip.message.Message;
+import org.zoolu.sip.message.MessageFactory;
+import org.zoolu.sip.provider.SipProvider;
+import org.zoolu.sip.transaction.TransactionClient;
+import org.zoolu.sip.transaction.TransactionClientListener;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Logger;
+
+
+public class PublishAgent implements TransactionClientListener
+{
+
+	/** UserProfile */
+	protected UserAgentProfile user_profile;
+	private Logger logger = Logger.getLogger(getClass().getCanonicalName());
+	/** SipProvider */
+	protected SipProvider sip_provider;
+	public String status;
+	//SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences();
+	//Boolean publish_enable_status  = prefs.getBoolean("publish_enable",true);
+	Boolean publish_enable_status=true;
+
+	/** Costructs a new MessageAgent. */
+	public PublishAgent(SipProvider sip_provider, UserAgentProfile user_profile)
+	{  this.sip_provider=sip_provider;
+		this.user_profile=user_profile;
+
+	}
+	public void publish()
+	{
+		status="open";

```

Comment:
this won't work. Make publish() be an voerloaded method which calls this.publish('open', 600, ''), where 600 is the default expiration time. This could be stored as a class attribute.